pkg/parquetquery/iters.go

```diff
@@ -123,36 +152,66 @@ func (t *RowNumber) Next(repetitionLevel, definitionLevel int) {
 			t[3] = -1
 			t[4] = -1
 			t[5] = -1
+			t[6] = -1
```

Comment:
i was noodling this and removing all the `t[N] = -1` from this function passes `TestSearchCompleteBlock`. This is b/c the row number is only `Next()'ed` in the iterators and it always receives the same definition level. So we are repeatedly (millions of times) setting those array positions to -1 for no reason. 

we could honestly change row number to be a slice and only have entries to the left of the required definition level.

Reply:
will follow up in another discovery/PR

Reply:
+1